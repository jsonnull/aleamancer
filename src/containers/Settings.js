// @flow
import React from 'react'
import { connect } from 'react-redux'
import Settings from 'components/Settings'
import { changeTheme, changeDisplayName } from 'actions'
import * as themes from 'styles/themes'
import type { ThemeName } from 'types'
import type { Theme } from 'styles/themes'
import type { State } from 'store'

export type StateProps = {
  theme: Theme,
  displayName: string,
  showSettings: boolean
}
const mapStateToProps = (state: State): StateProps => ({
  theme: themes[state.user.preferences.theme],
  displayName: state.user.profile.displayName,
  showSettings: state.ui.showSettings
})

export type DispatchProps = {
  changeTheme: (theme: ThemeName) => void,
  changeDisplayName: (name: string) => void,
  logout: Function,
  dismissSettings: Function
}
const mapDispatchToProps = (dispatch: Function): DispatchProps => ({
  changeTheme: (theme: ThemeName) => dispatch(changeTheme(theme)),
  changeDisplayName: (name: string) => dispatch(changeDisplayName(name)),
  logout: () => dispatch({ type: 'PERFORM_USER_LOGOUT' }),
  dismissSettings: () => dispatch({ type: 'HIDE_SETTINGS' })
})

export default connect(mapStateToProps, mapDispatchToProps)(Settings)
