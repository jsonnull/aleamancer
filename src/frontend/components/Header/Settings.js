// @flow
import React from 'react'
import Button from './Button'

type Props = {
  showSettings: Function
}

const Settings = (props: Props) => {
  const { showSettings } = props

  return (
    <Button square onClick={showSettings}>
      <i className="fa fa-cog" />
    </Button>
  )
}

export default Settings
