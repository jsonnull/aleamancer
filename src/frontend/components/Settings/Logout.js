// @flow
import React from 'react'
import styled from 'styled-components'
import Button from '../Button'

type Props = {
  logout: Function
}

const Container = styled.div`
  margin-top: 2rem;
  display: flex;
`

const Logout = (props: Props) => {
  return (
    <Container>
      <Button red onClick={props.logout}>
        Logout
      </Button>
    </Container>
  )
}

export default Logout
