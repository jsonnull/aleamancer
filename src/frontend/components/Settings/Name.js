// @flow
import React from 'react'
import styled from 'styled-components'
import Label from '../Label'
import Input from '../Input'
import editable from 'frontend/hoc/editable'

type Props = {
  name: string,
  onChange: Function
}

const Editable = editable(
  styled.div`
    position: relative;
    background: transparent;
    border: none;
    border-radius: 0;
    padding-left: 0;
    border-bottom: 1px solid ${props => props.theme.borderColor};
    color: ${props => props.theme.color};
    margin: 0 0 1rem;
    width: 100%;
    &:hover {
      cursor: pointer;
    }
  `
)

const NameInput = Input.extend`
  background: transparent;
  border: none;
  border-bottom: 1px solid ${props => props.theme.borderColor};
  color: ${props => props.theme.color};
  margin: 0 0 1rem;
  width: 100%;
  padding: 0;
`

const Name = (props: Props) => {
  return (
    <div>
      <Label>Display Name:</Label>
      <Editable
        isHovering={true}
        value={props.name}
        input={NameInput}
        onChange={props.onChange}
      />
    </div>
  )
}

export default Name
