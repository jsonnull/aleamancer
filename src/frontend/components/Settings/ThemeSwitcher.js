// @flow
import React from 'react'
import styled from 'styled-components'
import Label from '../Label'
import Button from '../Button'
import type { ThemeName } from 'common/types'

const DarkButton = Button.extend`
  margin-left: 5px;
`

const ThemeButtons = styled.div`
  display: flex;
`

type Props = {
  currentTheme: ThemeName,
  changeTheme: ThemeName => any
}

const ThemeSwitcher = (props: Props) => {
  const isLightTheme = props.currentTheme == 'light'
  const isDarkTheme = props.currentTheme == 'dark'

  const useGreen = { green: true }
  const useOutline = { outline: true }

  const lightButton = isLightTheme ? useGreen : useOutline
  const darkButton = isDarkTheme ? useGreen : useOutline

  return (
    <div>
      <Label>Theme</Label>
      <ThemeButtons>
        <Button {...lightButton} onClick={() => props.changeTheme('light')}>
          Light
        </Button>
        <DarkButton {...darkButton} onClick={() => props.changeTheme('dark')}>
          Dark
        </DarkButton>
      </ThemeButtons>
    </div>
  )
}

export default ThemeSwitcher
