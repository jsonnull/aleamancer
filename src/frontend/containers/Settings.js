// @flow
import { connect } from 'react-redux'
import Settings from 'frontend/components/Settings'
import { changeDisplayName, changeTheme } from 'frontend/actions'
import { HIDE_SETTINGS, PERFORM_USER_LOGOUT } from 'frontend/actions/types'
import displayNameSelector from 'frontend/selectors/displayName'
import * as themes from 'frontend/styles/themes'
import type { ThemeName } from 'common/types'
import type { Theme } from 'frontend/styles/themes'
import type { State } from 'frontend/store'

export type StateProps = {
  theme: Theme,
  displayName: string,
  showSettings: boolean
}
const mapStateToProps = (state: State): StateProps => ({
  theme: themes[state.preferences.theme],
  displayName: displayNameSelector(state),
  showSettings: state.ui.showSettings
})

export type DispatchProps = {
  changeTheme: (theme: ThemeName) => void,
  changeDisplayName: (name: string) => void,
  logout: Function,
  dismissSettings: Function
}
const mapDispatchToProps = (dispatch: Function) => ({
  changeTheme: (theme: ThemeName) => dispatch(changeTheme(theme)),
  changeDisplayName: (name: string) => dispatch(changeDisplayName(name)),
  logout: () => dispatch({ type: PERFORM_USER_LOGOUT }),
  dismissSettings: () => dispatch({ type: HIDE_SETTINGS })
})

export default connect(mapStateToProps, mapDispatchToProps)(Settings)
