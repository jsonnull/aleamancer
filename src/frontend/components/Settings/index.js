// @flow
import React from 'react'
import Modal from '../Modal'
import type { Theme } from 'frontend/styles/themes'
import type { DispatchProps, StateProps } from 'frontend/containers/Settings'
import Logout from './Logout'
import Name from './Name'
import ThemeSwitcher from './ThemeSwitcher'

type Props = {
  theme: Theme
} & StateProps &
  DispatchProps

const Settings = (props: Props) => {
  const {
    showSettings,
    displayName,
    changeDisplayName,
    theme,
    changeTheme,
    logout,
    dismissSettings
  } = props

  return (
    <Modal show={showSettings} dismiss={dismissSettings}>
      {() => (
        <div>
          <Name name={displayName} onChange={changeDisplayName} />

          <ThemeSwitcher currentTheme={theme.name} changeTheme={changeTheme} />

          <Logout logout={logout} />
        </div>
      )}
    </Modal>
  )
}

export default Settings
