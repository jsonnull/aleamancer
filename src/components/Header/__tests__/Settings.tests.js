// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import Settings from '../Settings'

describe('Header Settings component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Settings showSettings={() => {}} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
