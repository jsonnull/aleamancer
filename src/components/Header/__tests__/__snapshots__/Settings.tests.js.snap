// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Settings component renders correctly 1`] = `
<div
  className="sc-bdVaJa kgrFwP"
  onClick={[Function]}
>
  <i
    className="fa fa-cog"
  />
</div>
`;
